platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SignedStateFileWriter.java

```diff
@@ -135,27 +143,142 @@ public static void writeStateFile(final Path directory, final SignedState signed
+        copyPreconsensusEventStreamFiles(
+                platformContext,
+                selfId,
+                directory,
+                signedState.getState().getPlatformState().getPlatformData().getMinimumGenerationNonAncient());
+    }
+
+    /**
+     * Copy preconsensus event files into the signed state directory. These files are necessary for the platform to use
+     * the state file as a starting point. Note that starting with this only the PCES files in the state file does not
+     * guarantee that there is no data loss (i.e. there may be transactions that reach consensus after the state
+     * snapshot that are not replayed), but it does allow a node to start up and participate in gossip.
+     *
+     * <p>
+     * This general strategy is not very elegant is very much a hack. But it will allow us to do migration testing using
+     * real production states and streams, in the short term. In the longer term we should consider alternate and
+     * cleaner strategies.
+     *
+     * @param platformContext             the platform context
+     * @param destinationDirectory        the directory where the state is being written
+     * @param minimumGenerationNonAncient the minimum generation of events that are not ancient, with respect to the
+     *                                    state that is being written
+     */
+    private static void copyPreconsensusEventStreamFiles(
+            @NonNull final PlatformContext platformContext,
+            @NonNull final NodeId selfId,
+            @NonNull final Path destinationDirectory,
+            final long minimumGenerationNonAncient)
+            throws IOException {
+
+        final boolean copyPreconsensusStream = platformContext
+                .getConfiguration()
+                .getConfigData(PreconsensusEventStreamConfig.class)
+                .copyRecentStreamToStateSnapshots();
+        if (!copyPreconsensusStream) {
+            // PCES copying is disabled
+            return;
+        }
+
+        // The PCES files will be copied into this directory
+        final Path pcesDestination =
+                destinationDirectory.resolve("preconsensus-events").resolve(Long.toString(selfId.id()));
+        Files.createDirectories(pcesDestination);
+
+        // Gather all PCES files currently on disk.
+        final List<PreconsensusEventFile> allFiles = new ArrayList<>();
+        final Path preconsensusEventStreamDirectory =
+                PreconsensusEventFileManager.getDatabaseDirectory(platformContext, selfId);
+        try (final Stream<Path> stream = Files.walk(preconsensusEventStreamDirectory)) {
+            stream.filter(Files::isRegularFile).forEach(path -> {
+                try {
+                    allFiles.add(PreconsensusEventFile.of(path));
+                } catch (final IOException e) {
+                    // Ignore, this will get thrown for each file that is not a PCES file
+                }
```

Comment:
could we avoid this exception by adding a method like `isPcesFile`? I know this was done in a rush, so this is not a requirement

Reply:
If we do this, I'd prefer to do it as a stand alone PR. This is the same pattern used internally by the PCES file manager, and so if we change it here we probably want to change it in both places.

Normally I'd consider doing such things in a PR like this, but this is an urgent rush job that needs to be cherry picked ASAP.